collector/perfos.go

```diff
@@ -0,0 +1,132 @@
+// returns data points from Win32_PerfRawData_PerfOS_System class (undocumented)
+
+package collector
+
+import (
+	"log"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	Factories["perfos"] = NewPerfOSCollector
+}
+
+// A PerfOSCollector is a Prometheus collector for WMI metrics
+type PerfOSCollector struct {
+	ContextSwitchesPersec     *prometheus.Desc
+	ExceptionDispatchesPersec *prometheus.Desc
+	ProcessorQueueLength      *prometheus.Desc
+	SystemCallsPersec         *prometheus.Desc
+	SystemUpTime              *prometheus.Desc
+	Threads                   *prometheus.Desc
+}
+
+// NewPerfOSCollector ...
+func NewPerfOSCollector() (Collector, error) {
+	const subsystem = "perfos"
```

Comment:
No problem. Better get the naming right from the start, than having to change it later :)
I've renamed the file to system.go, the subsystem to "system". Patch updated.